### ISSUE TITLE
some function of MonsterHunterRise cant be hooked.

### ISSUE THREAD

**Issue by yun-wulian** [praydog/REFramework#825]
can comfirm: object "snow.player.PlayerQuestBase" 's function "getAdjustActionAttack" cant be hooked.
can confirm the function has been called by cheat engine's break point.

**Comment by praydog** [comment]
Upload your `re2_framework_log.txt` please.

**Comment by yun-wulian** [comment]
[re2_framework_log.zip](https://github.com/praydog/REFramework/files/11701732/re2_framework_log.zip)

**Comment by dtlnor** [comment]
This is from my end using `getAdjustActionAttack` hook
```
[2023-06-15 10:01:59.538] [REFramework] [info] [ScriptRunner] Module path C:\Program Files (x86)\Steam\steamapps\common\MonsterHunterRise\MonsterHunterRise.exe
[2023-06-15 10:01:59.538] [REFramework] [info] [ScriptRunner] Creating directories C:\Program Files (x86)\Steam\steamapps\common\MonsterHunterRise\reframework\autorun
[2023-06-15 10:01:59.538] [REFramework] [info] [ScriptRunner] Loading scripts...
[2023-06-15 10:01:59.538] [REFramework] [info] [ScriptState] Running script C:\Program Files (x86)\Steam\steamapps\common\MonsterHunterRise\reframework\autorun\HBGNormalLv3Cheat.lua...
[2023-06-15 10:01:59.539] [REFramework] [info] About to create a weak table for usertype int __cdecl sol_lua_push<struct sdk::RETypeDefinition>(struct sol::types<struct sdk::RETypeDefinition *>,struct lua_State *,struct sdk::RETypeDefinition *)
[2023-06-15 10:01:59.539] [REFramework] [info] Created a weak table for usertype int __cdecl sol_lua_push<struct sdk::RETypeDefinition>(struct sol::types<struct sdk::RETypeDefinition *>,struct lua_State *,struct sdk::RETypeDefinition *)
[2023-06-15 10:01:59.539] [REFramework] [info] About to create a weak table for usertype int __cdecl sol_lua_push<struct sdk::REMethodDefinition>(struct sol::types<struct sdk::REMethodDefinition *>,struct lua_State *,struct sdk::REMethodDefinition *)
[2023-06-15 10:01:59.539] [REFramework] [info] Created a weak table for usertype int __cdecl sol_lua_push<struct sdk::REMethodDefinition>(struct sol::types<struct sdk::REMethodDefinition *>,struct lua_State *,struct sdk::REMethodDefinition *)
[2023-06-15 10:01:59.569] [REFramework] [info] [HookManager] Using actual function @ 0x140cea310 for wrapper function @ 0x14065ee70
[2023-06-15 10:01:59.569] [REFramework] [info] [HookManager] Adding hook for 'getAdjustActionAttack' @ 0x140cea310...
[2023-06-15 10:01:59.569] [REFramework] [info] [HookManager] Reusing existing hook...
[2023-06-15 10:01:59.569] [REFramework] [info] [HookManager] Hook assigned ID 919
[2023-06-15 10:01:59.569] [REFramework] [info] [HookManager] Hook 919 added for 'getAdjustActionAttack' @ 0x140cea310
```

the log tells its hooking `0x140cea310` instead of wrapper function `0x14065ee70`.
`0x14065ee70` is `snow.player.PlayerQuestBase.getAdjustActionAttack(snow.hit.userdata.PlHitAttackRSData, snow.CharacterBase)` (a virtual function)
However `0x140cea310` is `snow.enemy.EnemyGrappleParam.clear()`
Which seems not related.

**Comment by dtlnor** [comment]
```cpp
void* get_actual_function(void* possible_fn) {
    if (possible_fn == nullptr) {
        return nullptr;
    }

    auto actual_fn = possible_fn;
    auto ip = (uintptr_t)possible_fn;

    // Disassemble the first few instructions to see if there is a jmp to an actual function.
    for (auto i = 0; i < 10; ++i) {
        hde64s hde{};
        auto len = hde64_disasm((void*)ip, &hde);
        ip += len;

        if (hde.opcode == 0xE9) { // jmp.
            actual_fn = (void*)(ip + hde.imm.imm32);
            break;
        }
    }

    if (possible_fn != actual_fn) {
        spdlog::info("[HookManager] Using actual function @ {:p} for wrapper function @ {:p}", actual_fn, possible_fn);
    }

    return actual_fn;
}
```
if I return the `possible_fn` directly, the method hook is worked.
Here is the instruction of this `getAdjustActionAttack` method
![image](https://github.com/praydog/REFramework/assets/21131439/46695baa-5777-4d37-86fd-f209d29c77ae)

**Comment by praydog** [comment]
A workaround to this is to set ignore_jmp parameter to true when calling `sdk.hook`. I'll see if I can modify `get_actual_function` to check if control flow ends abruptly.

**Comment by yun-wulian** [comment]
![U~1}36~4)PC@T2QABFES}~D](https://github.com/praydog/REFramework/assets/92302733/7215b235-de81-411f-952e-d5db64615a8f)
i think this isuue is because the function `get_actual_function` checking jmp(0xE9) in 10 lines of asm, but after `getAdjustActionAttack`  9 lines there's a jmp. so in `get_actual_function` maybe need a check for ret(0xC3).

**Comment by praydog** [comment]
Should be fixed in 322f513d581911749894153f5d1849eeafb71790
